pkg/block/local/adapter.go

```diff
@@ -165,6 +166,11 @@ func (l *Adapter) Put(_ context.Context, obj block.ObjectPointer, _ int64, reade
 	}()
 	_, err = io.Copy(f, reader)
 	if err != nil {
+		// Remove the file if it was partially written.
+		_ = os.Remove(p)
+		if errors.Is(err, syscall.ENOSPC) {
+			return nil, block.ErrStorageLimitExceeded
+		}
```

Comment:
1) A better handle here would upload the file to <file-name>.tmp and then move it to <file-name>.
This way, we won't delete an existing file in case we failed to upload the entire file.

2) I see no reason to return a different error here in case we got` syscall.ENOSPC`

Reply:
Sorry, for not being clear.
I didn't mean to upload to tmp dir, I meant to upload object.txt.tmp and then mv to object.txt.
Anyway, not a blocker for this PR, I just think it's a better way to handle uploads